File: 2024/Day 1/d01-easy.py
with open('input.txt', 'r') as f:
    lines = f.readlines()
    l,r=[], []
    for line in lines:
        nums = line.strip().split()
        if len(nums) == 2:  # Ensure there are exactly two numbers
            l.append(int(nums[0]))
            r.append(int(nums[1]))

l.sort()
r.sort()
d=0
for i in range(0, len(l)):
    d += abs(l[i] - r[i])
print(d)

File: 2024/Day 1/d01-easy.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <algorithm>
#include <cmath> // For std::abs

int main() {
    std::ifstream inputFile("input.txt");
    if (!inputFile.is_open()) {
        std::cerr << "Error: Could not open the file!" << std::endl;
        return 1;
    }

    std::vector<int> l, r;
    std::string line;

    // Read each line from the file
    while (std::getline(inputFile, line)) {
        std::istringstream lineStream(line);
        int num1, num2;
        
        // Parse two numbers separated by whitespace
        if (lineStream >> num1 >> num2) {
            l.push_back(num1);
            r.push_back(num2);
        }
    }

    inputFile.close();

    // Sort both lists
    std::sort(l.begin(), l.end());
    std::sort(r.begin(), r.end());

    // Calculate the total difference
    int d = 0;
    for (size_t i = 0; i < l.size(); ++i) {
        d += std::abs(l[i] - r[i]);
    }

    std::cout << d << std::endl;

    return 0;
}
